### ISSUE TITLE
Using a linear algebra library

### ISSUE THREAD

**Issue by garro95** [klangner/dsp#12]
What do you think of relying on an external linear algebra crate for the Vector representations and operations?
In this way, we could benefit from performance optimizations applied to the external crate.
Moreover, I think at a certain point we could face the need for Matrices (I am thinking, for example, to the implementation of the STFT) that for sure would be provided by the library.

**Comment by klangner** [comment]
Yes it makes sense. I would postpone it till Rust 2018 and we need linear algebra library with Complex number support.
